### PR TITLE
Fix Boolean abi inputs used in expressions

### LINKED ISSUE #668: Boolean Program Inputs errors out when used in If Expressions
# Description

## Aim

Program inputs of boolean type should behave no differently when used in if expressions. 

## Expected behavior

With `x = "1"` in _Prover.toml_, the following program should prove successfully:

```rust
fn main(x : bool) {
    if x {
        constrain 1 != 2;
    }
}
```

## Bug

Encountered the error below instead:

```
thread 'main' panicked at 'assertion failed: `(left == right)`
  left: `Boolean`,
 right: `Unsigned(1)`', crates/noirc_evaluator/src/ssa/node.rs:769:17
```

# Environment

For `nargo` users

- _noir-lang/noir_ commit cloned: [00b64a2d1f0838fad9df1aba5b15dd293cd80138](https://github.com/noir-lang/noir/commit/00b64a2d1f0838fad9df1aba5b15dd293cd80138)

### PR DESCRIPTION
**Pull request by vezenovm**
# Related issue(s)

Resolves #668

# Description

We were not correctly handling boolean inputs in the Abi. Under the hood boolean are always u1 integers however in SSA our NodeObj still differentiates between Boolean and Unsigned integer types. Thus, when we would transform a boolean abi input into an unsigned integer this would propagate all the way down to SSA. Once it came time to determine whether a boolean operation is possible this [assert](https://github.com/noir-lang/noir/blob/5d29fbc7370fd794307c04bf670eb65378104ff9/crates/noirc_evaluator/src/ssa/node.rs#L769) would fail when checking the types.

## Summary of changes

I added an AbiType for booleans. When parsing the Abi we now transform into a boolean type which gives us the correct `ObjectType` in SSA and we pass the assert.

## Test additions / changes

I expanded the `main_bool_arg` test when we enabled boolean inputs to use the code snippet in the linked issue.

# Checklist

- [X] I have tested the changes locally.
- [X] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` with default settings.
- [X] I have [linked](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) this PR to the issue(s) that it resolves.
- [X] I have reviewed the changes on GitHub, line by line.
- [X] I have ensured all changes are covered in the description.

# Additional context

### CODE REVIEW

File: crates/noirc_evaluator/src/ssa/conditional.rs [hunk 1]
```diff
@@ -168,7 +168,9 @@ impl DecisionTree {
         }
         let pvalue = self[assumption.parent].value.unwrap();
         let condition = self[assumption.parent].condition;
+        // dbg!(ctx.try_get_node(condition));
         let ins = if self.is_true_branch(block.assumption) {
+            // dbg!(ctx.try_get_node(pvalue));
```
Comment (jfecher) [inline crates/noirc_evaluator/src/ssa/conditional.rs:173]: Comments can be removed

**Review by jfecher — COMMENTED** [review]
This PR looks fine to me. I wonder if we should have both `u1` and `bool` in ssa at all though. I think we should only support `u1` and translate all bools into that instead to get rid of the duplication.

File: crates/noirc_evaluator/src/ssa/conditional.rs (hunk 1 above)
Comment (vezenovm) [reply crates/noirc_evaluator/src/ssa/conditional.rs]: Removed

**Comment by vezenovm** [comment]
> This PR looks fine to me. I wonder if we should have both `u1` and `bool` in ssa at all though. I think we should only support `u1` and translate all bools into that instead to get rid of the duplication.

I thought about this as well. The PR changes were the best quick fix so I did those for now. I can open a separate issue about removing the duplication as that seems like a separate refactor.

**Review by jfecher — APPROVED** [review]

**Pull request merged**